src/main/java/ca/on/oicr/gsi/visionmate/mockServer/MockScannerServer.java

```diff
@@ -17,67 +17,65 @@
 /**
  * Simple socket server to mock the behaviour of the VisionMate TCP/IP server for testing purposes. Intended to run in a thread:
  * <p>
+ * 
  * <pre>
- * {@code
- * MockScannerServer server = new MockScannerServer();
- * new Thread(server).start();
+ * {
+ *   &#64;code
+ *   MockScannerServer server = new MockScannerServer();
+ *   new Thread(server).start();
  * }
  * </pre>
  * <p>
- * When run, the server waits for a connection, will remain connected as long as the client stays, and will shut down when the 
- * connection is closed by the client.
+ * When run, the server waits for a connection, will remain connected as long as the client stays, and will shut down when the connection is
+ * closed by the client.
  * <p>
  * The mock server's state may be manipulated before issuing commands from a client.
  */
 public class MockScannerServer implements Runnable {
-  
+
   private final Logger log = Logger.getLogger(this.getClass());
-  
-  private static final int defaultPort = 8000;
-  
+
   private ServerConfig config = new ServerConfig();
-  private final int port;
-  
+  private final ServerSocket serverSocket;
+
   private RackType currentProduct = new RackType("M0812");
   private ScanStatus currentStatus = new ScanStatus(33); // initialized, rack96
   private Scan currentData = null;
-  
+
   /**
    * Constructs a new MockServerScanner that will listen on the default port (8000)
+   * 
+   * @throws IOException
    */
-  public MockScannerServer() {
-    this.port = defaultPort;
+  public MockScannerServer() throws IOException {
+    this(0);
   }
-  
+
   /**
    * Constructs a new MockServerScanner that will listen on the specified port
    * 
    * @param port
    */
-  public MockScannerServer(int port) {
-    this.port = port;
+  public MockScannerServer(int port) throws IOException {
+    serverSocket = new ServerSocket(port);
```

Comment:
This was previously declared in a try-with-resources block to take advantage of `AutoCloseable` and also to limit its opening. It would be preferable to leave it there, though with a randomized port, that means the port wouldn't be known until the thread was started. If it gets moved here, then it needs closed at some point